common/src/main/java/io/netty/util/concurrent/FastThreadLocalThread.java

```diff
@@ -71,6 +77,11 @@ public FastThreadLocalThread(ThreadGroup group, Runnable target, String name, lo
      * Note that this method is for internal use only, and thus is subject to change at any time.
      */
     public final InternalThreadLocalMap threadLocalMap() {
+        if (this != Thread.currentThread() && logger.isWarnEnabled()) {
+            logger.warn("It's not thread-safe to get 'threadLocalMap' which doesn't belong to the " +
+                            "caller thread, the stack trace of the caller thread is: {}",
+                    ThrowableUtil.stackTraceToString(new Exception()));
```

Comment:
Why not pass the cause directly as second argument ?

Reply:
Changed to log the exception directly with stack trace.